src/mapogcsld.cpp

```diff
@@ -196,6 +249,206 @@ static void msSLDApplySLD_DuplicateLayers(mapObj *map, int nSLDLayers,
 }
 #endif
 
+int msApplySldLayerToMapLayer(layerObj *sldLayer, layerObj *lp) {
+
+  if (sldLayer->numclasses > 0) {
+    int iClass = 0;
+    int k;
+    int bSLDHasNamedClass = MS_FALSE;
+
+    lp->type = sldLayer->type;
+    lp->rendermode = MS_ALL_MATCHING_CLASSES;
+
+    for (k = 0; k < sldLayer->numclasses; k++) {
+      if (sldLayer->_class[k]->group) {
+        bSLDHasNamedClass = MS_TRUE;
+        break;
+      }
+    }
+
+    for (k = 0; k < lp->numclasses; k++) {
+      if (lp->_class[k] != NULL) {
+        lp->_class[k]->layer = NULL;
+        if (freeClass(lp->_class[k]) == MS_SUCCESS) {
+          msFree(lp->_class[k]);
+          lp->_class[k] = NULL;
+        }
+      }
+    }
+    lp->numclasses = 0;
+
+    if (bSLDHasNamedClass && sldLayer->classgroup) {
+      /* Set the class group to the class that has UserStyle.IsDefault
+       */
+      msFree(lp->classgroup);
+      lp->classgroup = msStrdup(sldLayer->classgroup);
+    } else {
+      /*unset the classgroup on the layer if it was set. This allows the
+      layer to render with all the classes defined in the SLD*/
+      msFree(lp->classgroup);
+      lp->classgroup = NULL;
+    }
+
+    for (k = 0; k < sldLayer->numclasses; k++) {
+      if (msGrowLayerClasses(lp) == NULL)
+        return MS_FAILURE;
+
+      initClass(lp->_class[iClass]);
+      msCopyClass(lp->_class[iClass], sldLayer->_class[k], NULL);
+      lp->_class[iClass]->layer = lp;
+      lp->numclasses++;
+
+      /*aliases may have been used as part of the sld text symbolizer
+        for label element. Try to process it if that is the case #3114*/
+
+      int layerWasOpened = msLayerIsOpen(lp);
+
+      if (layerWasOpened || msLayerOpen(lp) == MS_SUCCESS) {
+        if (layerWasOpened || msLayerGetItems(lp) == MS_SUCCESS) {
```

Comment:
will require re-indenting the block below
```suggestion
      if (layerWasOpened || msLayerOpen(lp) == MS_SUCCESS || msLayerGetItems(lp) == MS_SUCCESS) {
```